src/interaction/selection.ts

```diff
@@ -18,15 +18,24 @@ export function start(): void {
 function recompute(): void {
     let s = document.getSelection();
 
-    if (selection !== null && data.start !== null && data.start !== getId(s.anchorNode)) {
+    // Bail out if we don't have a valid selection
+    if (s === null) { return; }
+
+    let anchorNode = getId(s.anchorNode);
+    let focusNode = getId(s.focusNode);
+
+    // Bail out if we got valid selection but not valid nodes
+    if (anchorNode === null && focusNode === null) { return; }
+
+    if (selection !== null && data.start !== null && data.start !== anchorNode) {
```

Comment:
Minor but you don't need the selection !== null here as you checked for it above.

Reply:
Technically what I've checked above is current selection (s), while this variable refers to last stored selection (selection). Since last is computed of current at some point - we don't need this additional check. I've also renamed the variables to make it clear.